### DISCUSSION TITLE
Disabling draw by the plugin

### DISCUSSION THREAD

**Discussion by joshting** [geoman-io/leaflet-geoman#1328] [Questions & Help]
I am sending the data to be persisted in the backend and only should draw the layer when I get the response back.  Therefore, I want to disable drawing by the plugin itself and draw it myself with the GeoJson value from the pm:create and pm:change events.

How do I disable the drawing by the plugin?  Or do I just delete it on those events?

Extending the question above.  Is it possible to create the point/line/polygon with leaflet-geoman hooks/events? i.e. programmatically create them after reading from cache/database/backend?  For example, I can get the GeoJson and using leaflet's L.geoJson(feature) to recreate it on map but it will no longer emit events like that draw using leaflet-geoman.

**Comment by Falke-Design** [comment]
You can disable the draw controls with [`setButtonDisabled(name, Boolean)`](https://github.com/geoman-io/leaflet-geoman#toolbar).

```
map.pm.Toolbar.setButtonDisabled('drawPolyline', true);
map.pm.Toolbar.setButtonDisabled('drawMarker', true);
map.pm.Toolbar.setButtonDisabled('drawCircle', true);
```

https://jsfiddle.net/758abqhy/

#### Importing GeoJSON and adding events:

You can use the default Leaflet [`onEachFeature`](https://leafletjs.com/reference.html#geojson-oneachfeature) option of `L.GeoJSON`:

```
L.geoJSON(geojson, {
   onEachFeature: function(feature, layer){
     layer.on('pm:change', console.log);
   }
});
```
